gcloud/datastore/test_client.py

```diff
@@ -18,29 +18,44 @@
 class TestClient(unittest2.TestCase):
 
     DATASET_ID = 'DATASET'
+    CONNECTION = object()
```

Comment:
`dataset_id` is supposed to be a string; `namespace` is supposed to be a string or None.  I don't like having the tests go outside those contracts.